gadgets/mri_core/generic_recon_gadgets/GenericReconBase.h

```diff
@@ -68,11 +72,140 @@ namespace Gadgetron {
         // exporter
         Gadgetron::ImageIOAnalyze gt_exporter_;
 
+        // store buffer names
+        std::map<std::string, std::pair<std::string, std::shared_ptr<std::ofstream> > > buffer_names_;
+
         // --------------------------------------------------
         // gadget functions
         // --------------------------------------------------
         virtual int process_config(ACE_Message_Block* mb);
         virtual int process(GadgetContainerMessage<T>* m1);
+        virtual int close(unsigned long flags = 1);
+
+        // --------------------------------------------------
+        // data stream functions
+        // --------------------------------------------------
+
+        void initialize_stream_name_buffer(const std::string& name);
+        void close_stream_buffer();
+
+        void stream_ismrmrd_header(const ISMRMRD::IsmrmrdHeader& hdr);
+
+        // stream of ND array buffer
+        template <typename DataType> 
+        void stream_to_array_buffer(const std::string& name, const hoNDArray<DataType>& data)
```

Comment:
All of these functions are not unique to this class. They could be a separate class which would give you the ability to test.

Reply:
Moved to the class implementation.